lib/improver/nowcasting/lightning.py

```diff
@@ -0,0 +1,455 @@
+        """
+        Modify the meta data of input cube to resemble a Nowcast of lightning
+        probability.
+        1. Rename to "lightning_probability"
+        2. Remove "threshold" coord
+
+        Args:
+            cube (iris.cube.Cube):
+                An input cube
+
+        Returns:
+            new_cube (iris.cube.Cube):
+                Output cube - a copy of input cube with meta-data relating to
+                a Nowcast of lightning probability.
+                The data array will be a copy of the input cube.data
+        """
+        new_cube = cube.copy()
+        new_cube.rename("lightning_probability")
+        new_cube.remove_coord('threshold')
+        if self.debug:
+            print(('In {}, new_cube is {}'.format(self, new_cube)))
+        return new_cube
+
+    def _modify_first_guess(self, cube, first_guess_lightning_cube,
+                            prob_lightning_cube, prob_precip_cube,
+                            prob_vii_cube):
+        """
+        Modify first-guess lightning probability with nowcast data
+
+        Args:
+            cube (iris.cube.Cube):
+                Provides the meta-data for the Nowcast lightning probability
+                output cube
+
+            first_guess_lightning_cube (iris.cube.Cube):
+                First-guess lightning probability
+                Must have same x & y dimensions as cube
+                Time dimension should overlap that of cube
+
+            prob_lightning_cube (iris.cube.Cube):
+                Nowcast lightning rate
+                Must have same dimensions as cube
+
+            prob_precip_cube (iris.cube.Cube):
+                Nowcast precipitation probability (threshold > 0.5, 7, 35)
+                Must have same other dimensions as cube
+
+            prob_vii_cube (iris.cube.Cube):
+                Radar-derived vertically integrated ice content (VII)
+                Must have same x and y dimensions as cube
+                Time should be a scalar coordinate
+                Must have a threshold coordinate with points matching
+                self.vii_thresholds
+                Can be <No cube> or None or anything that evaluates to False
+
+        Returns:
+            new_cube (iris.cube.Cube):
+                Output cube containing Nowcast lightning probability.
+        """
+        new_cube_list = iris.cube.CubeList([])
+        for cube_slice in cube.slices_over('time'):
+            this_point = cube_slice.coord('time').points[0]
+            this_time = iris_time_to_datetime(
+                cube_slice.coord('time').copy())[0]
+            prob_lightning_slice = prob_lightning_cube.extract(
+                iris.Constraint(time=this_time))
+            first_guess_time = iris_time_to_datetime(
+                first_guess_lightning_cube.coord('time').copy())[
+                    first_guess_lightning_cube.coord(
+                        'time'
+                        ).nearest_neighbour_index(this_point)]
```

Comment:
Why do we need a "nearest neighbour" here?  Are we expecting the lightning probability cube to have a different validity time from the other inputs?

Reply:
The data come from the UM and may not match in time. Doc-string is now clearer.